lib/src/component_declaration/builder_helpers.dart

```diff
@@ -126,102 +124,93 @@ abstract class UiProps extends component_base.UiProps with GeneratedClass {
       ' Will be removed in 4.0.0.')
   @toBeGenerated String get propKeyNamespace => throw UngeneratedError(member: #propKeyNamespace);
 
-  @toBeGenerated PropsInstanceMeta get $meta => throw UngeneratedError(member: #meta);
+  @toBeGenerated PropsMetaCollection get $meta => throw UngeneratedError(member: #meta);
 
   @override @toBeGenerated Map get props => throw UngeneratedError(member: #props);
 }
 
-mixin PropsConsumptionMixin on PropsMetaCollection {
-  Iterable<ConsumedProps> get defaultConsumedProps => all;
-
-  final List<ConsumedProps> _consumedProps = [];
-
-  Iterable<ConsumedProps> get consumedProps => _consumedProps.isEmpty ? defaultConsumedProps : _consumedProps;
+extension UiPropsMeta on UiProps {
+  static const consumedPropsAssertMessage = 'consumedProps must be either an Iterable<PropsMeta> or PropsMeta';
 
-  void consume(Type mixin) {
-    _consumedProps.add(forMixin(mixin));
-  }
+  /// A collection of metadata for the prop fields in all prop mixins used by
+  /// this props instance's generated props class.
+  ///
+  /// Synonymous with [UiComponent2]'s `propsMeta`.
+  ///
+  /// This can be used to derive consumed props by usage in conjunction with [addUnconsumedProps]
+  /// and [addUnconsumedDomProps].
+  PropsMetaCollection get meta => $meta;
 
-  void consumeMixins(Set<Type> mixins) {
-    _consumedProps.addAll(forMixins(mixins));
-  }
 
-  void consumeAllExceptFor(Set<Type> mixins) {
-    _consumedProps.addAll(allExceptForMixins(mixins));
-  }
+  /// Copies props from the provided [props] instance to this [UiProps] instance,
+  /// filtering out props found in [consumedProps].
+  ///
+  /// [consumedProps] should be either a [PropsMeta] or `Iterable<PropsMeta>` instance.
+  /// These are both return types of [PropsMetaCollection]'s related APIs
+  /// (`forMixin`, `forMixins`, `allExceptForMixins`).
```

Comment:
I think we should make it just `Iterable<PropsMeta>` to provide better typing and make it easier for consumers to tell if they're passing in the wrong thing.

The `forMixin` use-case should be able be changed to `forMixins` instead in most cases, or in edge cases wrap it in a list literal like we do in the consumedProps getter in components.

Reply:
Yep, good call, that is a lot cleaner!

Reply:
Addressed as part of ec43447!

Reply:
Consumers should prefer using `forMixins` instead of `forMixin`, (since that wraps it inside an iterable for them, and also makes it easier to add additional mixins later without accidentally duplicating them), so I don't think we should optimize for that use-case.

There's also the `.all` that can be used to get all of the mixins in an iterable.

So instead of:
```dart
final consumedProps = props.staticMeta.forMixin(ParentOnlyPropsMixin);
(SomeChild()..addUnconsumedProps(props, consumedProps.inList()))();
```
I think we should recommend:
```dart
final consumedProps = props.staticMeta.forMixins({ParentOnlyPropsMixin});
(SomeChild()..addUnconsumedProps(props, consumedProps))();
```

and for consumers who want to forward all props:
```dart
final consumedProps = props.staticMeta.all;
(SomeChild()..addUnconsumedProps(props, consumedProps))();
```

...and then we can eliminate `inList`.

Another benefit to that is that, `consumedProps` variable is typed as `Iterable<PropsMeta>`, just like the `consumedProps` getter is in class components, making it easier to translate between the two.

Reply:
Sweeet, that sounds good. I removed `inList` and updated the current examples in 501b2c8!